test/Util.test.js

```diff
@@ -0,0 +1,241 @@
+/* eslint-disable strict */
+const { Util, Collection } = require('../src');
+
+describe('Util unit tests', () => {
+  describe('Util#flatten', () => {
+    test('test collection flattening', () => {
+      const bar = new Collection();
+      bar.set('a', 1);
+      bar.set('b', 2);
+      bar.set('c', 3);
+
+      const obj = {
+        foo: 'foo',
+        bar,
+      };
+
+      expect(Util.flatten(obj)).toStrictEqual({ foo: 'foo', bar: ['a', 'b', 'c'] });
```

Comment:
This feels...wrong... why is bar an array of keys, and not an array of kv?

Reply:
because that is what Util#flatten does

Reply:
But is that really what we want it to do? CC @iCrawl @kyranet

Reply:
I don't understand your question. Please elaborate.

Reply:
I think they're suggesting that it it should flatten Collections into what would be `[key, value][]`

Reply:
How is that a concern of this PR?

Reply:
IMO we should keep this PR `semver: patch` or `semver: N/A`, and not promote it to `semver: major`. A change in the utilities can lead to unexpected bugs in the library.